sklego/preprocessing/dictmapper.py

```diff
@@ -0,0 +1,48 @@
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+from functools import partial
+
+
+class DictMapper(TransformerMixin, BaseEstimator):
+    """
+    Map the values of values of columns according to the input dictionary,
+    fall back to the default if the key is not present in the dictionary.
+
+    :param mapper: The dictionary containing the mapping of the values
+    :param default: The value to fall back to if the value is not in the mapper
+    """
+
+    def __init__(self, mapper, default):
+        self.mapper = mapper
+        self.default = default
+
+    def fit(self, X, y=None):
+        X = check_array(
+            X,
+            copy=True,
+            estimator=self,
+            force_all_finite=True,
+            dtype=None,
```

Comment:
about the dtype ... do we maybe want to exclude floats?

Reply:
I think this is fine for now actually. Just checked the documentation: the function might do an implicit conversion if this param is not `None`, which is iffy too.